cmd/create/cluster/cmd.go

```diff
@@ -2256,6 +2256,13 @@ func run(cmd *cobra.Command, _ []string) {
 	}
 
 	if isSubnetBelongToSharedVpc(r, awsCreator.AccountID, subnetIDs, mapSubnetIDToSubnet) {
+
+		if clusterHasLongNameWithoutDomainPrefix(clusterName, domainPrefix) {
+			r.Reporter.Errorf("Installing a cluster into shared VPC is only supported for cluster " +
+				"which has a name no longer than 15 characters or with domain prefix")
+			os.Exit(1)
+		}
```

Comment:
Let's also involve the docs team to modify the part around cluster DNS in following doc: https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa-shared-vpc-config.html

instead of;

```
Create a private hosted zone in the [Route 53 section of the AWS console](https://us-east-1.console.aws.amazon.com/route53/v2/). In the hosted zone configuration, the domain name is <cluster_name>.<reserved_dns_domain>. The private hosted zone must be associated with the created VPC.
``` 

it should be
```
Create a private hosted zone in the [Route 53 section of the AWS console](https://us-east-1.console.aws.amazon.com/route53/v2/). In the hosted zone configuration, the domain name is <cluster_domain_prefix>.<reserved_dns_domain>. The private hosted zone must be associated with the created VPC.
```
Also instead of 
```
After you create the Operator roles, share the full domain name, which is created with <intended_cluster_name>.<reserved_dns_domain>, your Ingress Operator Cloud Credentials role’s ARN, and your Installer role’s ARN with the VPC Owner to continue configuration.
```

it should be 
```
After you create the Operator roles, share the full domain name, which is created with <intended_cluster_domain_prefix>.<reserved_dns_domain>, your Ingress Operator Cloud Credentials role’s ARN, and your Installer role’s ARN with the VPC Owner to continue configuration.
```

Reply:
thanks @machi1990 
I will create a doc issue to make the change